semantic-conventions/src/opentelemetry/semconv/model/semantic_attribute.py

```diff
@@ -116,7 +120,7 @@ def parse(
                 msg = "At least one of id or ref is required."
                 raise ValidationError.from_yaml_pos(position, msg)
             if attr_id is not None:
-                validate_id(attr_id, position_data["id"])
+                validate_attribute_id(attr_id, position_data["id"])
                 attr_type, brief, examples = SemanticAttribute.parse_id(attribute)
                 if prefix:
                     fqn = f"{prefix}.{attr_id}"
```

Comment:
The key name now is part of the fqn. This is IMHO wrong. The fqn, for the purpose of uniqueness-checks and referencing should be independent of the key name. E.g. it should be invalid to have both `request.header.<key>` and `request.header.<name>`, or even both `request.header` and `request.header.<key>`.

I think it will be cleaner to implement if the id is just the ID of the namespace and the fact that it's a template is indicated with some other way.

If you want to stay with the indicator in the ID, consider using `*` instead of `<customkeyname>`, and adding special handling to catch the case when both `name` and `name.*` exist.

Reply:
> I think it will be cleaner to implement if the id is just the ID of the namespace and the fact that it's a template is indicated with some other way.

But isn't that exactly what I proposed before and @lmolkova was concerned about:

> otherwise, we're defining a namespace request.header under attributes section and it does not feel right

I'm really fine with both. Even just keeping the `id` as `request.header`. The implication of that would be that `request.header` is identifying this set of attributes, but it cannot be used as a namespace for other attributes, which is fine IMHO.

Reply:
The more I think about it, the more I like your idea, @Oberon00 

> Another idea would be to use `type: namespace[string]` or similar, but that would be difficult to implement I think.

Reply:
> But isn't that exactly https://github.com/open-telemetry/build-tools/pull/186#issuecomment-1628321598 and @lmolkova was concerned about?:

I think it is. @lmolkova is this a big concern or just a nice-to-(not-)have?

Reply:
Agree, we can validate namespace vs attribute name reuse with tooling given we have a special template type, no concern from my side.